### PR TITLE
Feature/gauge chart

### PR DESCRIPTION
**Pull request by nxanil**
@ronakmshah @curran 

I am also done with the Gauge chart for review (not yet completed due to waiting for elasticsearch query). This is how currently it looked like at my side:
screenshot from 2017-04-11 19-11-41

For now, I have used the "aar-flow-sla-heatmap" query, but in general, I am not using it at all inside the Gauge Chart. 

Here is the code that basically used for fetching the Range and Current Value in GaugeGraph/index.js:
`       const minValue         = data[minColumn] ? data[minColumn] : 0;
        const maxValue         = data[maxColumn] ? data[maxColumn] : 100;
        const currentValue     = data[currentColumn] ? data[currentColumn] : 20;`

So basically these are not defined in the data returned by the elastic-search, as a result, it uses default values i.e. 20 and 100 respectively.

Once you will provide me with the exact elastic search query, I will update the code accordingly.

### CODE REVIEW

File: src/components/Graphs/GaugeGraph/index.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+        const currentValue     = data[currentColumn] ? data[currentColumn] : 20;
+
+        const minRadius   = Math.min(availableWidth, availableHeight) / 2;
+
+        const innerRadius = minRadius - gaugeRingInset - gaugeRingWidth;
+        const outerRadius = minRadius - gaugeRingInset;
+
+        const pointerHeadLength = Math.round(minRadius * gaugePtrHeadLengthPercent);
+
+        const arcColor = d3.interpolateHsl(d3.rgb(colors[0] ? colors[0] : '#e8e2ca'), d3.rgb(colors[1] ? colors[1] : '#3e6c0a'))
```
Comment (curran) [inline src/components/Graphs/GaugeGraph/index.js]: Very nice.

File: src/components/Graphs/GaugeGraph/index.js [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+        const range = angles.max - angles.min;
+
+        const arc = d3.arc()
+    			  .innerRadius(innerRadius)
```
Comment (curran) [inline src/components/Graphs/GaugeGraph/index.js]: There's a spacing issue here with tabs vs. spaces. Please use spaces rather than tabs, to be consistent with the rest of the codebase (you can see this if you look at the [diff](https://github.com/nuagenetworks/visualization-framework/pull/359/files)). Thanks!

File: src/components/Graphs/GaugeGraph/index.js [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+
+    		let ticks = scale.ticks(gaugeTicks);
+        let tickData = d3.range(gaugeTicks).map(function() {return 1 / gaugeTicks;});
+
+        const textLabel = ((d) => {
+            const formatter = d3.format(labelFormat || ",.2s");
+            return formatter(d);
+        });
+
+        const transformText = ((d, i) => {
```
Comment (curran) [inline src/components/Graphs/GaugeGraph/index.js]: Really nice approach here.

**Comment by nxanil** [comment]
@curran 
Indentation issues are fixed now for gauge chart.

**Comment by curran** [comment]
@nxanil Great, thanks for the change. I'd say this is ready for merge.

@ronakmshah Would you like to do the merge, or should I?
